### ISSUE TITLE
Question: Alpine versus alternative base images

### ISSUE THREAD

**Issue by srabraham** [burningmantech/ranger-clubhouse-web#1459]
I see that the web and api Clubhouse Docker images use Alpine as a base OS. I'm wondering if alternatives have been considered, e.g. Debian slim or Ubuntu or something else.

More context:
* Usually people choose Alpine because it's super small. That's really the only thing it has going for it. Debian slim isn't significantly larger (maybe a few hundred MB, vs. 5 MB for Alpine).
* Alpine uses the musl C library, while almost all other Linux distros use glibc. Not only does this imply a different between production and development environments, but also it leads to the next point:
* Alpine has been known to have worse performance in applications due to that musl vs. glibc difference. Interesting reads here: https://snyk.io/blog/choosing-the-best-node-js-docker-image/ and https://pythonspeed.com/articles/alpine-docker-python/

Anyway, I'm curious about any background here, and whether it'd make sense to experiment with Debian slim, Ubuntu, or something else. @mikeburg @flwyd

**Comment by mikeburg** [comment]
@srabraham Please move this conversation over to the Tech Ninja's email list. 

The  Clubhouse is not a high traffic'ed website. Most of the year it's idle with maybe 1 to 2 people visiting per day during the winter months. When the operations are in full swing out on playa,  you'll see about 10 to 20 APIs per second during shift changes, each request completing within 10 to 40 ms. Even then, the bottleneck is the database, not the Laravel instance.

In other words, we don't need to worry about  any potential optimization gained through switching distros at this time.

**Comment by srabraham** [comment]
Thanks for the response, Mike. Emailed. I maintain some skepticism about Alpine as a whole here, in part for software supportability's sake. e.g. NodeJS even seems to [recommend caution with Alpine](https://hub.docker.com/_/node) due to that abnormal libc package it uses. I may investigate this a bit more in the future, when I have more than a few days' worth of experience with the Clubhouse code :).

Further conversation can be done on the email thread.

**Comment by wsanchez** [comment]
The only possibly compelling argument I can think of for switching would be faster build times, if, in fact, they are faster, just to speed up CI. That could be tested in a PR.
